src/org/labkey/test/LabKeySiteWrapper.java

```diff
@@ -1265,6 +1265,11 @@ public void stopImpersonating()
      */
     public void stopImpersonating(Boolean goHome)
     {
+        if (!onLabKeyClassicPage()) // Single-page apps don't have impersonation capabilities
+        {
+            goToHome();
+            goHome = false;
+        }
```

Comment:
I'd argue against making this change for a couple of reasons:

Presumably since the developer had to navigate to home, and leave the single page app, to start impersonating that is the first hint that they would have to go back to stop impersonating.

This change no longer honors the value of goHome that has been passed in which makes the function misleading. If it is set to false the developer gets unexpected behavior (they were navigated to home even if they said not to).

If you have to leave this in I would suggest that it calls stopImpersonatingHTTP() if not in LabKey classic.

Or alternately could you get the current URL before going home and then navigate back? You could do the same with impersonate which would make the (navigation) behavior of impersonate and stopImpersonating the same in the single page app.

Reply:
That's reasonable. I made this change so that the post-test validation and cleanup didn't crap out if a test left off impersonating in a single-page app. I'll move this to the `stopImpersonation` line of `ensureSignedInAsPrimaryTestUser`, which is explicitly intended to be a blunt instrument.